Workflow JavaScript Sample -  Parse DQL result and trigger conditional branches

Dynatrace-Tomislav opened the pull request:
DT Workflow template / sample that shows how to achieve the following:

1) Parse the result of a DQL query in JavaScript (iterate through result JSON array).
For quick testing: DQL query (first task) generates dummy data with DQL ‘data’ command

2) Based on the the processed result from 1), then trigger conditional branches (Jinja expressions). For example: 
- If production error count > 100 ---> branch Workflow into task A
- If production error count >= 0 and <= 100 ---> branch Workflow into task B
- If production namespace not found ---> branch Workflow into task C

![Workflow_Sample_JavaScript_Parse_DQL](https://github.com/Dynatrace/Dynatrace-workflow-samples/assets/14933193/a2580465-350e-44d1-91c4-51efd5413e78)

agrimmer reviewed (commented):
Thank you for the PR! The use case is interesting! 
However, there seems to be a simpler solution that does not require any custom Javascript. I rebuilt the Workflow with a modified DQL and modified Conditions. I've attached it to this review 
[wf_conditions_sample_413b7276-627a-459b-aea5-f5e2678b0ef9.json](https://github.com/Dynatrace/Dynatrace-workflow-samples/files/14901901/wf_conditions_sample_413b7276-627a-459b-aea5-f5e2678b0ef9.json)
What is your opinion on this alternative solution?

Dynatrace-Tomislav commented:
> Thank you for the PR! The use case is interesting! However, there seems to be a simpler solution that does not require any custom Javascript. I rebuilt the Workflow with a modified DQL and modified Conditions. I've attached it to this review [wf_conditions_sample_413b7276-627a-459b-aea5-f5e2678b0ef9.json](https://github.com/Dynatrace/Dynatrace-workflow-samples/files/14901901/wf_conditions_sample_413b7276-627a-459b-aea5-f5e2678b0ef9.json) What is your opinion on this alternative solution?

Hi @agrimmer! Hi Andreas! Thanks a lot for the review! The shorter Workflow version without any custom JavaScript is great! It is very elegant to achieve the same result for this use case! Maybe this could be added as an additional sample?

However, I added my first, longer Workflow version intentionally with JavaScript to share an example of a template if any further, more complex custom logic / processing of the DQL result is needed within in the Workflow task. One could then use this longer Workflow template version (with JavaScript) and then expand it with more complex logic / processing with JS, if needed.

In summary, my suggestion is: Could we have both WF versions as samples side by side? 
1) The shorter version without any custom JavaScript (which purely evaluates the result of the DQL query in a much shorter way) and
2) The longer WF version with JavaScript (intentionally) for educational purposes as explained above (to have a sample template if more complex custom logic / processing of the DQL result is needed, etc)

Dynatrace-Tomislav commented on samples/JavaScript - Parse DQL Result and Trigger Conditional Branches/wf_workflow_javascript_-_parse_dql.json:
```diff
@@ -0,0 +1,106 @@
+      },
+      "predecessors": []
+    },
+    "process_dql_result": {
```
Yes, I also added the same DQL filter in the first DQL WF task (it is commented out). I agree that one would usually want to filter directly in the DQL query itself as you explained. The reason why I added JavaScript for the filter here is purely for educational purposes, to share an example of a template if any further logic/processing with JavaScript needs to be done in the Workflow. Both options are in the Workflow sample (commented out in DQL and explained with comments in the JS task)

Dynatrace-Tomislav commented on samples/JavaScript - Parse DQL Result and Trigger Conditional Branches/wf_workflow_javascript_-_parse_dql.json:
```diff
@@ -0,0 +1,106 @@
+      "description": "Process result of DQL query (from previous task)",
+      "action": "dynatrace.automations:run-javascript",
+      "input": {
+        "script": "// optional import of sdk modules\nimport { execution } from '@dynatrace-sdk/automation-utils';\n\nexport default async function ({ execution_id }) {\n\n  // --------------------------------------------\n  // Step 1) Get result from previous DQL task\n  // --------------------------------------------\n  console.log(\"get result from previous task...\");\n  var task_info = await ...
```
Thanks a lot for the suggestion! The shorter SDK alternative is better.

agrimmer commented:
I like your suggestion to add both samples. May I ask you to add it to this PR and also adapt the Readme?
Then, I can approve the PR.

Dynatrace-Tomislav commented:
Hi @agrimmer! I have added the second sample (version without JavaScript) to the pull request, and I have updated the Readme. I have also updated the first sample (with JavaScript) to use the SDK to get the result (as suggested above). It has now both samples.

Could you please review and then merge the Pull Request? 

Thanks a lot in advance!

agrimmer commented:
Thank you!
LGTM
